App/screens/ProofRequest.tsx

```diff
@@ -19,115 +25,144 @@ interface Props {
-    })
-  }
-  return transformedAttributes
-}
-
 const CredentialOffer: React.FC<Props> = ({ navigation, route }) => {
   const { agent } = useAgent()
-  const [buttonsVisible, setButtonsVisible] = useState(true)
-  const [retrievedCredentials, setRetrievedCredentials] = useState<RetrievedCredentials>(null)
-  const [retrievedCredentialsDisplay, setRetrievedCredentialsDisplay] = useState<any>(null)
-  const proofId = route?.params?.proofId
-  const proof = useProofById(proofId)
-  const connection = useConnectionById(proof?.connectionId)
   const { t } = useTranslation()
+  const [buttonsVisible, setButtonsVisible] = useState(true)
 
-  useEffect(() => {
-    if (proof?.state === ProofState.Done) {
+  const [retrievedCredentials, setRetrievedCredentials] = useState<RetrievedCredentials | null>(null)
+  const [retrievedCredentialsDisplay, setRetrievedCredentialsDisplay] = useState<CredentialDisplay[] | null>(null)
+
+  const transformAttributes = (attributes: Record<string, RequestedAttribute[]>): CredentialDisplay[] => {
+    const transformedAttributes = []
+    for (const attribute in attributes) {
+      transformedAttributes.push({
+        name: attribute,
+        value: attributes[attribute][0].credentialInfo.attributes[attribute],
+        credentialDefinitionId: parseSchema(attributes[attribute][0].credentialInfo.schemaId),
+      })
+    }
+    return transformedAttributes
+  }
+
+  const getProofRecord = (proofId?: string): ProofRecord | void => {
+    try {
+      if (!proofId) {
+        throw new Error(t('ProofRequest.ProofNotFound'))
+      }
+      return useProofById(proofId)
+    } catch (e: unknown) {
+      // console.error(e)
       Toast.show({
-        type: 'success',
-        text1: t('ProofRequest.SuccessfullyAcceptedProof'),
+        type: 'error',
+        text1: (e as Error)?.message || t('Global.Failure'),
       })
       navigation.goBack()
     }
-  }, [proof])
+  }
 
-  const getRetrievedCredentials = async () => {
+  const getConnectionRecordFromProof = (connectionId?: string): ConnectionRecord | void => {
+    if (connectionId) {
+      return useConnectionById(connectionId)
+    }
+  }
+
+  const getRetrievedCredentials = async (proof: ProofRecord) => {
     try {
-      if (!proof?.requestMessage?.indyProofRequest) {
-        Toast.show({
-          type: 'error',
-          text1: t('ProofRequest.ProofNotFound'),
-        })
-        throw new Error('Indy proof request not found')
-      }
-      const retrievedCreds = await agent?.proofs?.getRequestedCredentialsForProofRequest(
-        proof?.requestMessage?.indyProofRequest,
-        undefined
-      )
-      if (!retrievedCreds) {
-        Toast.show({
-          type: 'error',
-          text1: t('ProofRequest.RequestedCredsNotFound'),
-        })
-        throw new Error('Retrieved creds not found')
+      const creds = await agent?.proofs?.getRequestedCredentialsForProofRequest(proof?.id)
```

Comment:
Might be good to look into a way where we wouldn't need all those `?`

I.e do a check once and have type safety afterwards